src/callbacks/liquidity/Ramses/RamsesV2DTL.sol

```diff
@@ -0,0 +1,239 @@
+    ///             - the callback has `baseTokenAmount_` quantity of base tokens
+    function _mintAndDeposit(
+        uint96 lotId_,
+        address quoteToken_,
+        uint256 quoteTokenAmount_,
+        address baseToken_,
+        uint256 baseTokenAmount_,
+        bytes memory callbackData_
+    ) internal virtual override {
+        // Decode the callback data
+        OnSettleParams memory params = abi.decode(callbackData_, (OnSettleParams));
+
+        // Extract the pool fee from the implParams
+        uint24 poolFee = abi.decode(lotConfiguration[lotId_].implParams, (uint24));
+
+        // Determine the ordering of tokens
+        bool quoteTokenIsToken0 = quoteToken_ < baseToken_;
+
+        // Create and initialize the pool if necessary
+        {
+            // Determine sqrtPriceX96
+            uint160 sqrtPriceX96 = SqrtPriceMath.getSqrtPriceX96(
+                quoteToken_, baseToken_, quoteTokenAmount_, baseTokenAmount_
+            );
+
+            // If the pool already exists and is initialized, it will have no effect
+            // Please see the risks section in the contract documentation for more information
+            _createAndInitializePoolIfNecessary(
+                quoteTokenIsToken0 ? quoteToken_ : baseToken_,
+                quoteTokenIsToken0 ? baseToken_ : quoteToken_,
+                poolFee,
+                sqrtPriceX96
+            );
+        }
+
+        // Mint the position and add liquidity
+        {
+            IRamsesV2PositionManager.MintParams memory mintParams = _getMintParams(
+                lotId_, quoteToken_, quoteTokenAmount_, baseToken_, baseTokenAmount_, params
+            );
+
+            // Approve spending
+            ERC20(quoteToken_).approve(address(ramsesV2PositionManager), quoteTokenAmount_);
+            ERC20(baseToken_).approve(address(ramsesV2PositionManager), baseTokenAmount_);
+
+            // Mint the position
+            ramsesV2PositionManager.mint(mintParams);
+
+            // Reset dangling approvals
+            // The position manager may not spend all tokens
+            ERC20(quoteToken_).approve(address(ramsesV2PositionManager), 0);
+            ERC20(baseToken_).approve(address(ramsesV2PositionManager), 0);
+        }
+    }
+
+    /// @inheritdoc BaseDirectToLiquidity
+    /// @dev        This function does not perform any actions,
+    ///             as `_mintAndDeposit()` directly transfers the token to the recipient
+    function _transferPoolToken(uint96) internal virtual override {
+        // Nothing to do
+    }
+
+    // ========== INTERNAL FUNCTIONS ========== //
+
+    /// @dev    Copied from UniswapV3's PoolInitializer (which is GPL >= 2)
+    function _createAndInitializePoolIfNecessary(
+        address token0,
+        address token1,
+        uint24 fee,
+        uint160 sqrtPriceX96
+    ) internal returns (address pool) {
+        require(token0 < token1);
```

Comment:
this check isn't necessary since you check this before the function, but doesn't hurt